### DISCUSSION TITLE
Many compressor start/stops during normal heating

### DISCUSSION THREAD

**Discussion by FlexxFR** [Arnold-n/P1P2MQTT#135] [Q&A] [answered]
Now I have full insight with this P1P2 & ESPAltherma modules I'm trying to learn how this thing actually works :)
I see now during a "normal" heating cycle that the heatpump is going trough many on/off cycles.

Example:
![image](https://github.com/user-attachments/assets/a7440a93-fc07-4b0a-8545-0d618341e425)

Top graph shows the floorwater temp, going up- and down.
Compressor start/stop explains itself, while the mode was still heating
bottom 2 graphs, are the power (external measurement) and waterflow, confirming the on/off mode of the heatpump.

I read everywhere to limit the number of starts/stops. Is this normal behaviour? I feel it's not tuned properly but I have NO CLUE :) how this works. There are so many settings. Has this something to do with "modulation?"

**Any suggestions how to improve?**

Maybe these config settings are relevant ;)
![image](https://github.com/user-attachments/assets/5185d625-21b9-45ce-be71-3e598e688dd1)

**Comment by Arnold-n** [comment, marked as answer]
> I read everywhere to limit the number of starts/stops. Is this normal behaviour? I feel it's not tuned properly but I have NO CLUE :) how this works. There are so many settings. Has this something to do with "modulation?"

Your system appears to operate at minimum power yet still overshoots  its target LWT. This is indeed better avoided in view of wear and efficiency. Setting overshoot to 4, and restarting your Daikin, may delay or avoid such overshoot. Increasing target LWT may also help.

It has nothing to do with room-modulation as that is off.

**Reply by FlexxFR** [reply]
after some weeks of testing it much more stable now.

**Comment by FlexxFR** [comment]
Hi Arnold, thanks for your fast reply.
Is assume RT/LWT is controlling temperature based on Room Thermostat of Leaving Water, right? RT is fin I guess.

Increasing target LWT, is that related to the temperature graph line (based on outside temperatures)?
In general lower LWT should be more efficient, right?  But in my case the current (probably low LWT) is so low that even with minimal power (1kW), its still to much. Am I right?

I will first try to set the overshoot form 1 to 4 and see what happens.

**Reply by Arnold-n** [reply]
Indeed, usually LWT is determined by the weather dependent curve based on the (possibly averaged) outside temperature. Alternatively you can operate in absolute LWT mode and directly determine target LWT. This applies both in room-temperature mode and in LWT mode. If your house thermal reaction is slow enough, I would recommend 24h averaging to have stable low power operation. Strangely this requires that you switch energy saving of the outside unit off or averaging will not work.

**Comment by Staacks** [comment]
Just a few more thoughts from my experience of fixing extreme cycling for my setup:

- Overshoot is indeed a good and quick fix. Just be aware that setting overshoot as high as 4 is very similar to increasing your target LWT as it will often be several degrees above your original target. (Meaning less efficient than it could be and/or rooms get warmer than you might want them to be). You might want to try and get a stable system at an overshoot of only 2 instead - if you can.
- The first and most important thing you should check is the hydronic balancing of your system. In particular, if there is a path through your heating system that has a much higher flow than other paths without actually heating a large part of your house, then this flow path acts like a short circuit: The hot water will flow through this path without getting much colder and just go back as hot returning water, causing your heat pump to think that not that much heat is needed and to eventually shut it off. It will cycle on and off and you still have a difficult time to heat your house through the rest of your heating system. So a proper hydronic balancing might be one of the most important parts for smooth operation.
- Single room thermostats are a bad idea. They will suddenly open or close and change the behavior of your entire system. Even worse, too many of them might close at the same time, restricting the flow through your system too much. This can also lead to cycling behavior as the few remaining rooms simply cannot take the heating power of your heatpump even if it is modulated to the minimum. Do not use single room thermostats, but find a good constant equilibrium for the entire house. At least try opening all of them to see if this fixes your issue. (Heating from a subset of rooms is also less efficient in a well insulated house where rooms will reach the same temperature anyway.)
- If the outside temperature is rather warm or your house has been heated from sunlight, some cycling might be unavoidable. The house is warm and simply will not take the heat from your heatpump. And I am sure that this is fine if it only occurs on warm days as Daikin will have factored this in. I would love to have a less aggressive setting from Daikin, but I think they prefer this cycling behavior to satisfy customers with "immediate" heating systems like active heatpump convectors, where you would notice a temperature drop very fast if the heatpump just stays off longer. Anyhow, if you still want to avoid this and have a "slow" heating system (i.e. underfloor heating where it takes hours to even notice that the heating is off), you can set up a statistics sensor in HA to count the number of starts within a few hours. Then set up an automation that turns off heating for three hours (timer entity) whenever the pump flow goes to zero while the number of starts already is too high (triggering on the pump dropping to zero prevents this automation from interrupting a startup). Don't forget to turn it on again when the timer runs out :)

**Reply by DiabloVmax1200** [reply]
Hi,
the last bullet point should be possible to solve if you use the controlling by room temperature. I also did not try it yet because I'm in the optimization phase with thermal balancing. But I've read in a German Board that Daikin recommends to use this setting to reduce the on/off cycles.

**Comment by FlexxFR** [comment]
After the change it already looks better, although I'm still not sure what the "downside" is of this change. Will this lead to an overshoot in room temperature?

![image](https://github.com/user-attachments/assets/e86f92f6-f920-4843-aaf2-6237a52a15e7)


Second bullet:
Not sure how to do this. I only have floor heating so it's a very slow process. While I'm thinking, my systems does have 3 seperate circulation pumps, one for each floor. Second floor + attic do not need to be heated so I switch these pumps off. I measure intake/outtake temperatures on each floor and what I see is that the same temp water is returned. Not sure if all these headers (one per floor) each have their own main line back to the heatpump. In that case I have 2 lines "shorted". Does this makes sense?

Bullet three:
I only have one thermostat (that Daikin panel), requesting heat or not. No single room valves. What I do use are "floor" thermostats (via home assistant, average floor temp and switch off the entire floor circulation pump. (Actually, these floors are not yet heated past weeks, not cold enough).

Bullet 4: 
If it's "hot" enough outside, why should the thermostat go to "heating" in the first place? 


anyway, thanks for your answers.
Interesting stuff to know a bit more how this thing actually works.

**Reply by Staacks** [reply]
> After the change it already looks better, although I'm still not sure what the "downside" is of this change. Will this lead to an overshoot in room temperature?

I have to admit that I was thinking of my weather-dependent LWT set up, not room-temperature mode. I would guess, that room temperature is not a problem then. Maybe efficiency due to the slightly higher temperatures.

> Second bullet: Not sure how to do this. I only have floor heating so it's a very slow process. While I'm thinking, my systems does have 3 seperate circulation pumps, one for each floor. Second floor + attic do not need to be heated so I switch these pumps off. I measure intake/outtake temperatures on each floor and what I see is that the same temp water is returned. Not sure if all these headers (one per floor) each have their own main line back to the heatpump. In that case I have 2 lines "shorted". Does this makes sense?

If shutting off the floors actually means bypassing them, then this might explain your problem. But it is hard to say without knowing your setup and I only have experience with my setup here. At least I can describe the problem I had: In my house the first and second floor have floor heating, but the basement had simple radiators (which wasn't even a good idea with gas heating). When I got my heatpump installed, I also got two heatpump convectors replacing the radiators. The problem was that they did not include a shut-off valve, so when they were turned off, the water was running through them without losing much heat and right back into the heatpump (which actually is almost next door in the basement). Took me a few days to figure it out and luckily there was also a ball valve that allowed to cut off the entire line to the convectors in the basement. When I closed that, the problem vanished immediately, convincing my installer to add a shut-off valve to each convector.

So, if you have a valve to close each floor, you might want to try that to see what happens. However, the answer to the next question might play a role, too...

> 
> Bullet three: I only have one thermostat (that Daikin panel), requesting heat or not. No single room valves. What I do use are "floor" thermostats (via home assistant, average floor temp and switch off the entire floor circulation pump. (Actually, these floors are not yet heated past weeks, not cold enough).

Again, hard to tell if this is a problem, but I can tell what I saw in my setup: We had valves for each room, which might make a some sense if you use gas heating with rather high temperatures. But in a well insulated house, the rooms still end up with similar temperatures as they are not as well instulated to each other as to the outside. When using a heatpump, you want to use as much surface with the lowest possible temperature to heat your house, so shutting off one room and having the neighboring room compensate for that is not really what you would want. But that is just the efficiency point of view.

The bigger problem that might also affect you, is that the heatpump requires a minimum flowrate. I am not entirely sure why, but the manual clearly states that if you have valves to shut down some parts, you need to measure the flowrate in the worst-case situation of all valves being closed. If the flowrate goes too low, it will throw an error. There is a test mode in the installers part of the menus to measure the flow rate. So, if you want to try shutting off the other floors, you might want to use that to see if you are still above the required flowrate.

Not sure what to recommend here. If the flow through a single floor is high enough, stopping the flow through the others might help (IF that even is the problem). If the flow is not enough, heating all floors might be an option if your house is well insulated and the heat from the first floor might just go up to the cooler floors anyway. There is also the option of a bypass valve for situations with low flowrates, but I am not really sure if that would be any better than what you have now.

> Bullet 4: If it's "hot" enough outside, why should the thermostat go to "heating" in the first place?

Again, sorry, I was thinking of my LWT setup. In this case the rooms can be a bit different from the weather that governs the LWT. In case of a RT-driven system, it is probably easier to stabilize it by using higher water temperatures and letting the thermostat do it's job. The only problem only occurs if the water temperature is so close too the room temperature. (But you want to get as close as possible for that sweet COP :) )

**Comment by Arnold-n** [comment]
Thanks for the very useful overview, @Staacks, and fully agree, though I see no reason why the LWT would remain higher than the target due to the higher overshoot - unless the minimum power operation dictates a higher temperature anyway. Especially when the heat pump just starts after being off, or after a defrost, it seems to operate shortly at a higher power setting (the graph  shows 2.5kW), and if it then overshoots (target-LWT+overshoot+0.9C) it will cycle and not reach a continuous low power operation. This is a known problem for hybrid systems where the overshoot is fixed at 1C.

As for heating when it is warm outside: sometimes there are short periods during the day when it is nice and warm outside, but the house can still be too cold overall - it is very efficient to use these warmer periods to heat if the house has enough thermal capacity, and with dynamic prices these moments may also be more economical to heat.

**Reply by Staacks** [reply]
Jepp, sorry, I was mixing my weather-dependent LWT setup with FlexxFR's RT setup. The thermostat should still do its job.

However, increasing the overshoot temperature can be very similar to increasing the LWT, depending on the reason for the overshoot. My heatpump also starts with higher power and slowly adjusts to match the LWT setpoint. I sometimes see it start several times in a short rapid sequence before it gets lucky to not overshoot and then it runs perfectly for the rest of the day. That's why I increased the overshoot setting from Daikin's default of 1°C.

But in situations when it cannot go to lower powers, the overshoot also means that it will keep running until the overshoot is exceeded. That can easily be several hours of the LWT slowly creeping up to setpoint+overshoot and that is several hours of an increased LWT.

Here is what my heatpump did over the last few hours with an overshoot of 2°C: 
![image](https://github.com/user-attachments/assets/62d5c5bd-b9ab-4335-b7ac-9f1964ab9fb6)

That's almost six hours during which it was not able to keep the LWT (red) on the setpoint (green). It started about 1°C higher than it should be and the heatpump decided that it is time to give up only when it hit almost 3°C above the setpoint.

But as mentioned above with an RT setup, this will probably not affect room temperature, but efficiency.
